### PR TITLE
MINOR: Adjust `testClientDisconnectionUpdatesRequestMetrics` to also test small response case

### PR DESCRIPTION
**Pull request by ijuma**
The test also passes with small responses now in both macOS Catalina and Linux.
Remove outdated comment and exercise both paths.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Two jobs passed, one failed with two unrelated flaky test failures

**Comment by ijuma** [comment]
retest this please

**Comment by ijuma** [comment]
cc @junrao @rajinisivaram

File: core/src/test/scala/unit/kafka/network/SocketServerTest.scala [hunk 1]
```diff
@@ -744,15 +744,10 @@ class SocketServerTest {
 
       val requestMetrics = channel.metrics(request.header.apiKey.name)
       def totalTimeHistCount(): Long = requestMetrics.totalTimeHist.count
-      val expectedTotalTimeCount = totalTimeHistCount() + 1
+      val send = new NetworkSend(request.context.connectionId, ByteBuffer.allocate(0))
```
Comment (junrao) [inline core/src/test/scala/unit/kafka/network/SocketServerTest.scala]: While you are at it, it seems that we need to make conn volatile?

**Review by junrao — APPROVED** [review]
@ijuma : Thanks for the PR. LGTM. One minor comment below. Also, I turned on the debug logging and it seems that the server detects the closed socket through read.

```
    [2019-11-27 11:49:12,527] DEBUG Connection with /127.0.0.1 disconnected (org.apache.kafka.common.network.Selector:601)
    java.io.IOException: Connection reset by peer
        at java.base/sun.nio.ch.FileDispatcherImpl.read0(Native Method)
        at java.base/sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
        at java.base/sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:276)
        at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:245)
        at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:223)
        at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:358)
        at org.apache.kafka.common.network.PlaintextTransportLayer.read(PlaintextTransportLayer.java:103)
        at org.apache.kafka.common.network.NetworkReceive.readFrom(NetworkReceive.java:95)
        at org.apache.kafka.common.network.KafkaChannel.receive(KafkaChannel.java:443)
        at org.apache.kafka.common.network.KafkaChannel.read(KafkaChannel.java:394)
        at org.apache.kafka.common.network.Selector.attemptRead(Selector.java:675)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:572)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:482)
        at kafka.network.Processor.poll(SocketServer.scala:890)
        at kafka.network.Processor.run(SocketServer.scala:789)
        at java.base/java.lang.Thread.run(Thread.java:835)

```

**Comment by ijuma** [comment]
@junrao That's right for a small buffer. If the buffer is large, it detects it via an IOException on write (connection reset). Is it worth having a test for that too?

**Comment by ijuma** [comment]
Btw, the interesting thing is that the behavior changed since I wrote the test. I don't know if the macOS networking stack changed, our code or something else. :)

**Comment by junrao** [comment]
When sending a large response, I got the following. So, the broken socket was detected through writes.

```
    [2019-11-27 13:06:20,950] DEBUG Connection with /127.0.0.1 disconnected (org.apache.kafka.common.network.Selector:601)
    java.io.IOException: Broken pipe
        at java.base/sun.nio.ch.FileDispatcherImpl.writev0(Native Method)
        at java.base/sun.nio.ch.SocketDispatcher.writev(SocketDispatcher.java:51)
        at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:182)
        at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:130)
        at java.base/sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:496)
        at java.base/java.nio.channels.SocketChannel.write(SocketChannel.java:507)
        at org.apache.kafka.common.network.PlaintextTransportLayer.write(PlaintextTransportLayer.java:152)
        at org.apache.kafka.common.network.ByteBufferSend.writeTo(ByteBufferSend.java:60)
        at org.apache.kafka.common.network.KafkaChannel.write(KafkaChannel.java:422)
        at org.apache.kafka.common.network.Selector.write(Selector.java:639)
        at org.apache.kafka.common.network.Selector.attemptWrite(Selector.java:632)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:588)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:482)
        at kafka.network.Processor.poll(SocketServer.scala:890)
        at kafka.network.Processor.run(SocketServer.scala:789)
        at java.base/java.lang.Thread.run(Thread.java:835)

```

**Comment by ijuma** [comment]
Yes, exactly. That's what I got too.

**Comment by ijuma** [comment]
Is it worth having a test for this path too?

**Comment by junrao** [comment]
Yes, I guess it won't hurt to exercise the large send code path too.

**Review by andrewchoi5 — APPROVED** [review]

**Review by rajinisivaram — APPROVED** [review]
@ijuma Thanks for the PR, LGTM

**Pull request merged**